src/index.js

```diff
@@ -313,145 +289,174 @@ function addSegment(currentState, segment) {
 
 // The main interface
 
-export var RouteRecognizer = function() {
-  this.rootState = new State();
-  this.names = {};
-};
-
-
-RouteRecognizer.prototype = {
-  add: function(routes, options) {
-    var currentState = this.rootState, regex = "^",
-        types = { statics: 0, dynamics: 0, stars: 0 },
-        handlers = [], allSegments = [], name;
-
-    var isEmpty = true;
-
-    for (var i=0, l=routes.length; i<l; i++) {
-      var route = routes[i], names = [];
-
-      var segments = parse(route.path, names, types);
+export class RouteRecognizer {
+  constructor() {
+    this.map = map;
+    this.rootState = new State();
+    this.names = {};
+  }
 
-      allSegments = allSegments.concat(segments);
+  add(route) {
+    if (Array.isArray(route)) {
+      for (let r of route) {
+        this.add(r);
+      }
 
-      for (var j=0, m=segments.length; j<m; j++) {
-        var segment = segments[j];
+      return;
+    }
 
-        if (segment instanceof EpsilonSegment) { continue; }
+    var currentState = this.rootState, regex = '^',
+        types = { statics: 0, dynamics: 0, stars: 0 },
+        names = [], routeName = route.handler.name,
+        isEmpty = true;
 
-        isEmpty = false;
+    var segments = parse(route.path, names, types);
+    for (let segment of segments) {
+      if (segment instanceof EpsilonSegment) {
+        continue;
+      }
 
-        // Add a "/" for the new segment
-        currentState = currentState.put({ validChars: "/" });
-        regex += "/";
+      isEmpty = false;
 
-        // Add a representation of the segment to the NFA and regex
-        currentState = addSegment(currentState, segment);
-        regex += segment.regex();
-      }
+      // Add a '/' for the new segment
+      currentState = currentState.put({ validChars: '/' });
+      regex += '/';
 
-      var handler = { handler: route.handler, names: names };
-      handlers.push(handler);
+      // Add a representation of the segment to the NFA and regex
+      currentState = addSegment(currentState, segment);
+      regex += segment.regex();
     }
 
     if (isEmpty) {
-      currentState = currentState.put({ validChars: "/" });
-      regex += "/";
+      currentState = currentState.put({ validChars: '/' });
+      regex += '/';
     }
 
-    currentState.handlers = handlers;
-    currentState.regex = new RegExp(regex + "$");
-    currentState.types = types;
+    var handlers = [{ handler: route.handler, names: names }];
```

Comment:
Is it possible to have multiple handlers for a state? If so, how? Need tests. If not, we should remove the array.